hibernate-core/src/main/antlr/hql-sql.g

```diff
@@ -436,7 +436,7 @@ selectExpr
 	;
 
 count
-    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR ) ) {
+    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR | query) ) {
```

Comment:
@sebersole  I have nothing against supporting this but I ignore the problems you and @gbadner faced with subquery in the past so my opinion may be wrong.